src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -736,6 +742,22 @@ public static bool IsSingleton(string set)
                 return false;
         }
 
+        /// <summary>
+        /// <c>true</c> if the set contains a single character only
+        /// </summary>
+        public static bool IsDoubleton(string set)
+        {
+            if (set[FLAGS] == 0 && // it's not negated. [a-z^aeiuo] shows negation
```

Comment:
have you checked if you could use the IsNegated helper function here and get rid of the set == null check there?

Reply:
I can certainly do that and reuse in IsSingleton. I was trying to minimise the change. Perhaps I can defer that cleanup for 2.2...?